### PR TITLE
ICU: bump version

### PR DESCRIPTION
**Pull request by kenmays**
Bumped ICU74 to v74.2
Closes: https://github.com/haikuports/haikuports/issues/10650

Ref: https://icu.unicode.org/download/74

"ICU 74.2 maintenance release
Release date: 2023-12-13

ICU 74.2 updates to [CLDR 44.1](https://cldr.unicode.org/index/downloads/cldr-44#h.nvqx283jwsx) locale data. These are maintenance releases for ICU 74 and CLDR 44, with limited sets of bug fixes and no API or structural changes.

The CLDR bug fix relevant for ICU is for some formatting patterns that erroneously had two adjacent space characters. These are coalesced into one. ([CLDR-17233](https://unicode-org.atlassian.net/browse/CLDR-17233))

[List of tickets fixed in ICU 74.2](https://unicode-org.atlassian.net/projects/ICU/versions/10336/tab/release-report-all-issues)

Important: DateFormat.getInstanceForSkeleton() and the DateTimePatternGenerator sometimes used the wrong patterns because they failed to use/inherit certain data. ([ICU-22575](https://unicode-org.atlassian.net/browse/ICU-22575) — CLDR 44 had removed some redundant data that ICU relied on.)"

### CODE REVIEW

**Comment by Begasus** [comment]
Why remove the older version?? We're not there yet to clean all those up.

**Comment by kenmays** [comment]
We only need ICU 57, 66, 74.
- R1B4 - 57, 66
- Nightly x86_gcc2: 57, 74
- Nightly x86, x64, riscV: 74
- Nightly arm, arm64, sparc, ppc: 57

**Comment by Begasus** [comment]
For Haiku, haikuports still contain some that use the older ICU versions.

**Comment by kenmays** [comment]
I've readded all of the ones used by the build system or Haikuports.

**Comment by Begasus** [comment]
Please keep in the older ones for now, they will be removed when the time is right and checking has been done for those.

**Comment by Begasus** [comment]
Looking again this will over rule icu74 in any way, obsoleting current icu74 (which is needed for Haiku). Let's keep it open for a while.

File: dev-libs/icu/icu74-74.2.recipe [hunk 1]
```diff
@@ -39,54 +39,55 @@ providing very competitive performance.
 a range of text, or identify locations that would be suitable for line \
 wrapping when displaying the text."
 HOMEPAGE="http://www.icu-project.org"
-COPYRIGHT="1995-2020 IBM Corporation and others."
+COPYRIGHT="2016-2024 Unicode, Inc."
 LICENSE="ICU"
 REVISION="1"
 SOURCE_URI="https://github.com/unicode-org/icu/releases/download/release-${portVersion/\./-}/icu4c-${portVersion/\./_}-src.tgz"
-CHECKSUM_SHA256="818a80712ed3caacd9b652305e01afc7fa167e6f2e94996da44b90c2ab604ce1"
+CHECKSUM_SHA256="e6c7876c0f3d756f3a6969cad9a8909e535eeaac352f3a721338b9cbd56864bf7414469d29ec843462997815d2ca9d0dab06d38c37cdd4d8feb28ad04d8781b0"
```
Comment (Begasus) [inline dev-libs/icu/icu74-74.2.recipe:46]: How did you come up with this checksum?
```
Error: Expected SHA-256: e6c7876c0f3d756f3a6969cad9a8909e535eeaac352f3a721338b9cbd56864bf7414469d29ec843462997815d2ca9d0dab06d38c37cdd4d8feb28ad04d8781b0
Error: Found SHA-256:    68db082212a96d6f53e35d60f47d38b962e9f9d207a74cfac78029ae8ff5e08c
```

File: dev-libs/icu/icu74-74.2.recipe [hunk 2]
```diff
@@ -39,54 +39,55 @@ providing very competitive performance.
...
 SOURCE_URI_2="https://github.com/unicode-org/icu/releases/download/release-${portVersion/\./-}/icu4c-${portVersion/\./_}-data.zip#noarchive"
-CHECKSUM_SHA256_2="ca1ee076163b438461e484421a7679fc33a64cd0a54f9d4b401893fa1eb42701"
+CHECKSUM_SHA256_2="f9dbd303f78de1bf9089262211f3b618f1ec915e57877855d0bc6496332620f4ea92eabe1dff9fa721600e9a6ce56885f79361bbcdf97d0cfedd18e4a2d58ad0"
```
Comment (Begasus) [inline dev-libs/icu/icu74-74.2.recipe:48]: Same here:
```
Error: Expected SHA-256: f9dbd303f78de1bf9089262211f3b618f1ec915e57877855d0bc6496332620f4ea92eabe1dff9fa721600e9a6ce56885f79361bbcdf97d0cfedd18e4a2d58ad0
Error: Found SHA-256:    c28c3ca5f4ba3384781797138a294ca360988d4322674ad4d51e52f5d9b0a2b6
```

**Comment by waddlesplash** [comment]
We should never upgrade ICU by removing an old version.

**Pull request closed without merging**